projects/plugins/jetpack/extensions/blocks/one-payment/edit.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { get, map } from 'lodash';
+import classnames from 'classnames';
+import { __ } from '@wordpress/i18n';
+import { useEffect } from '@wordpress/element';
+import { compose, withInstanceId } from '@wordpress/compose';
+import { createBlock, registerBlockVariation } from '@wordpress/blocks';
+import { withDispatch, withSelect } from '@wordpress/data';
+import {
+	InnerBlocks,
+	__experimentalBlockVariationPicker as BlockVariationPicker,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import defaultVariations from './variations';
+
+export function JetpackOnePaymentEdit( {
+	setAttributes,
+	hasInnerBlocks,
+	replaceBlock,
+	selectBlock,
+	clientId,
+	className,
+	blockType,
+	variations,
+	defaultVariation,
+} ) {
+	const formClassnames = classnames( className, 'jetpackone-payment' );
+
+	const createBlocksFromInnerBlocksTemplate = innerBlocksTemplate => {
+		const blocks = map( innerBlocksTemplate, ( [ name, attr, innerBlocks = [] ] ) =>
+			createBlock( name, attr, createBlocksFromInnerBlocksTemplate( innerBlocks ) )
+		);
+
+		return blocks;
+	};
+
+	const setVariation = variation => {
+		if ( variation.attributes ) {
+			setAttributes( variation.attributes );
+		}
+
+		if ( variation.innerBlocks ) {
```

Comment:
Is it possible for a variation to not have inner blocks?
AFAICS the variation list always add inner blocks for each variations (as long as the block is registered):
`innerBlocks: [ [ 'jetpack/' + blockName, {} ] ],`